src/main/java/frc/robot/util/Blinkin.java

```diff
@@ -24,25 +29,71 @@ private static Spark getController() {
    * @param tableInput numerical input for Pattern/Color to use
    */
   public static void setColor(double tableInput) {
+    isDualColor = false;
     currColor = tableInput;
   }
   /**
    * sets pattern/color for Blinkin using Blinkin enum so you don't need to reference the table for numerical ids in the documentation
    * @param color Pattern/Color to use
    */
   public static void setColor(Colors color) {
+    isDualColor = false;
     currColor = color.m_id;
   }
 
+  /**
+   * Sets colors for dual color mode. These colors will be cycled on the Blinkin every 600 milliseconds
+   * @param color1 First color to be displayed
+   * @param color2 Second color to be displayed
+   */
+  public static void setDualColor(Colors color1, Colors color2) {
+    isDualColor = true;
+    dualColors = new double[] {color1.m_id, color2.m_id};
+  }
+
+  /**
+   * Blinks a single color on the LEDs.
+   * @param color
+   */
+  public static void blinkColor(Colors color) {
+    setDualColor(color, Colors.BLACK);
+  }
+
+  /**
+   * Blinks team colors, orange and black, on Blinkin. Black will turn LEDs off, so really only orange is shown.
+   */
+  public static void blinkTeamColors() {
+    setDualColor(Colors.ORANGE, Colors.BLACK);
+  }
+
+  /**
+   * Periodic method for cycling colors on Blinkin. Checks if timer has passed 0.5 seconds, and if it has
+   * color will be changed.
+   */
+  public static void dualColorPeriodic() {
+    if (m_timer.hasElapsed(0.5)) {
+      m_timer.reset();
+      if (dualColorIndex == 0) {
+        getController().set(dualColors[1]);
+      } else if (dualColorIndex == 1) {
+        getController().set(dualColors[0]);
+      }
+    }
```

Comment:
I think you need to increment dualColorIndex

Reply:
Oh yeah, good catch! I think I've fixed it now.